### PR TITLE
Fix dropdown unexpected closing

### LINKED ISSUE #1998: dropdown is closing by click on inner link/button
`dropdown_switcher_{link,button}` is closing by click on inner link/button as it subscribes on switcher by class.

### PR DESCRIPTION
**Pull request by tadatuta**
Closes #1998

---

Showcase is [available](http://bem.github.io/reports/6570/showcase/showcase.html)

### CODE REVIEW

File: common.blocks/dropdown/dropdown.bemhtml.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 block('dropdown')(
     replace()(function() {
-        return [{ elem : 'popup' }, { elem : 'switcher' }];
+        return [
+            { elem : 'popup' },
+            { elem : 'switcher', elemMods : { switcher : this.mods.switcher } }
```
Comment (veged) [inline common.blocks/dropdown/dropdown.bemhtml.js]: we can automatically proxy `this.mods.switcher` in `dropdown__switcher.bemhtml.js`

File: common.blocks/dropdown/dropdown.js [hunk 2]
```diff
@@ -86,7 +86,10 @@ provide(bemDom.declBlock(this.name, /** @lends dropdown.prototype */{
         this.setMod('opened', data.modVal);
     }
 }, /** @lends dropdown */{
-    lazyInit : true
+    lazyInit : true,
+    onInit : function() {
+        this._events(Switcher).on('click', this.prototype.onSwitcherClick);
```
Comment (veged) [inline common.blocks/dropdown/dropdown.js]: in base declaration there is no `this.prototype.onSwitcherClick` — maybe we need default stub with `throw` about mandatory use some value of `_switcher` modifier

**Review by veged — CHANGES_REQUESTED** [review]

**Comment by dfilatov** [comment]
lgtm

File: common.blocks/dropdown/dropdown.js (hunk 2 above)
Comment (dfilatov) [inline common.blocks/dropdown/dropdown.js]: let's rename `onSwitcherClick` to `_onSwitcherClick`

Comment (tadatuta) [reply common.blocks/dropdown/dropdown.js]: actually there is: https://github.com/bem/bem-components/blob/96736547f914964c1652db26560f987b241b6611/common.blocks/dropdown/dropdown.js#L80

**Comment by tadatuta** [comment]
added fixes in a separate commit

**Pull request merged**